nsqd/nsqd.go

```diff
@@ -51,7 +55,8 @@ type NSQD struct {
 	errValue  atomic.Value
 	startTime time.Time
 
-	topicMap map[string]*Topic
+	topicMap  map[string]*Topic
+	producers map[int64]Producer
```

Comment:
Calling this variable `producers` seems misleading because we can't really know or categorize for sure. I'd rather call this `clients`.